src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -535,22 +536,20 @@ def specify_all_power_straps_by_tracks(self, layer_names: List[str], ground_net:
 
         :param layer_names: The list of metal layer names on which to create straps.
         :param ground_net: The name of the ground net in this design. Only 1 ground net is supported.
-        :param power_nets: A list of power nets to create (not ground). Currently only supports 1 (e.g. ["VDD"]).
+        :param power_nets: A list of power nets to create (not ground).
         :param power_weights: Specifies the power strap placement pattern for multiple-domain designs (e.g. ["VDDA", "VDDB"] with [2, 1] will produce 2 VDDA straps for ever 1 VDDB strap).
         :param bbox: The optional (2N)-point bounding box of the area to generate straps. By default the entire core area is used.
         :param pin_layers: A list of layers on which to place pins
         :return: A list of TCL commands that will generate power straps.
         """
         assert len(power_nets) == len(power_weights)
-        if len(power_nets) > 1:
-            raise NotImplementedError("FIXME: I don't yet support multiple power domains")
 
         # Do some sanity checking
         for l in pin_layers:
             assert l in layer_names, "Pin layer {} must be in power strap layers".format(l)
 
-        # TODO when implementing multiple power domains, this needs to change based on the floorplan
-        output = self.specify_std_cell_power_straps(bbox, [ground_net, power_nets[0]])
+        # TODO does the CPF help this, or do we need to be more explicit about the bbox for each domain
```

Comment:
Its going to be hard to get all possible power domain structures to work but for some of the ones we've already created manually the CPF can help along with the pin api.

Reply:
Will file an issue to track this, but I agree it'll be hard.